idunn/api/places_list.py

```diff
@@ -228,33 +190,61 @@ async def get_places_bbox_impl(
     )
 
 
-async def _fetch_places_list(params: PlacesQueryParam, poi_es_index_name: str):
-    if params.source == PoiSource.PAGESJAUNES:
-        return await run_in_threadpool(
-            pj_source.get_places_bbox,
-            params.category,
-            params.bbox,
-            size=params.size,
-            query=params.q,
-        )
-    if params.q:
-        # Default source (OSM) with query
-        bragi_response = await bragi_client.pois_query_in_bbox(
-            query=params.q, bbox=params.bbox, lang=params.lang, limit=params.size
-        )
-        return [BragiPOI(f) for f in bragi_response.get("features", [])]
+async def select_datasource(params):
+    if pj_source.bbox_is_covered(params.bbox):
+        await select_datasource_for_france(params)
+    else:
+        await select_datasource_outside_france(params)
 
-    # Default source (OSM) with category or class/subclass filters
-    if params.raw_filter:
-        filters = [MimirPoiFilter.from_url_raw_filter(f) for f in params.raw_filter]
+
+async def select_datasource_for_france(params):
+    if any(cat in params.category for cat in TRIPADVISOR_CATEGORIES_COVERED_IN_FRANCE):
```

Comment:
I'd argue that this `any` should be `all` if we want to be sure to cover all categories with the selected datasource. I don't think it actually matters because for a regular use of Qwant Maps this `category` parameter should always be a list of exactly one element?

Reply:
If we use `all` to cover all categories, I think that I should reverse the expression : `all(cat in TRIPADVISOR_CATEGORIES_COVERED_IN_FRANCE for cat in params.category)`